### PR TITLE
Api

### PR DESCRIPTION
**Pull request by Tommy-FF**

### CODE REVIEW

File: app/src/main/java/de/dhbw/corona_world_app/api/APIManager.java [hunk 1]
```diff
@@ -133,21 +133,30 @@ public Country call() throws Exception {
 
     public static List<TimeframedCountry> getData(List<ISOCountry> countryList, List<Criteria> criteriaList, LocalDate startDate, LocalDate endDate) throws ExecutionException, InterruptedException, JSONException {
         Logger.logV(TAG, "Getting data according to following parameters: " + countryList + " ; " + criteriaList);
+        if((startDate == null && endDate != null) || startDate.isBefore(endDate)) throw new IllegalArgumentException("Ending date is before starting date!");
```
Comment (aleksn41) [inline app/src/main/java/de/dhbw/corona_world_app/api/APIManager.java]: if endDate==null and startDate!=null causes an NullPointer exception

**Review by aleksn41 — CHANGES_REQUESTED** [review]

File: app/src/main/java/de/dhbw/corona_world_app/api/APIManager.java (hunk 1 above)
Comment (aleksn41) [reply app/src/main/java/de/dhbw/corona_world_app/api/APIManager.java]: or when startDate==null and endDate==null

Comment (Tommy-FF) [reply app/src/main/java/de/dhbw/corona_world_app/api/APIManager.java]: Yeah, will fix that. I stumbled upon it already :)

File: app/src/main/java/de/dhbw/corona_world_app/api/APIManager.java [hunk 2]
```diff
@@ -126,28 +130,50 @@ public Country call() throws Exception {
...
         return returnList;
     }
 
-    public static List<TimeframedCountry> getData(List<ISOCountry> countryList, List<Criteria> criteriaList, LocalDate startDate, LocalDate endDate) throws ExecutionException, InterruptedException, JSONException {
+    public static List<TimeframedCountry> getData(@NonNull List<ISOCountry> countryList, @NonNull List<Criteria> criteriaList, LocalDate startDate, LocalDate endDate) throws ExecutionException, InterruptedException, JSONException {
         Logger.logV(TAG, "Getting data according to following parameters: " + countryList + " ; " + criteriaList);
+        if ((startDate == null && endDate != null))
+            throw new IllegalArgumentException("Ending date is before starting date!");
+        if (startDate != null && endDate != null && endDate.isBefore(startDate))
+            throw new IllegalArgumentException("Ending date is before starting date!");
```
Comment (aleksn41) [inline app/src/main/java/de/dhbw/corona_world_app/api/APIManager.java]: Same error, put it into one if

File: app/src/main/java/de/dhbw/corona_world_app/api/APIManager.java [hunk 3]
```diff
@@ -126,28 +130,50 @@ public Country call() throws Exception {
...
+        final LocalDate finalStartDate;
+        final LocalDate finalEndDate = endDate;
+
+        //this is only needed because the api cannot handle when start and end date are equal and returns all dates' data
+        boolean startAndEndEqual = startDate.equals(endDate);
+        if(startAndEndEqual){
+            finalStartDate = startDate.minusDays(1);
+        } else {
+            finalStartDate = startDate;
+        }
```
Comment (aleksn41) [inline app/src/main/java/de/dhbw/corona_world_app/api/APIManager.java:157]: Finalstart- and enddate seem to be useless variables here

**Review by aleksn41 — CHANGES_REQUESTED** [review]

File: app/src/main/java/de/dhbw/corona_world_app/api/APIManager.java (hunk 3 above)
Comment (Tommy-FF) [reply app/src/main/java/de/dhbw/corona_world_app/api/APIManager.java:157]: no, they have to be declared this way or else they cannot be passed to the executor service

**Pull request merged**